lib/orogen/gen/tasks.rb

```diff
@@ -533,6 +533,20 @@ def generate
                     hidden_operation("__orogen_getTID", "    #ifdef HAS_GETTID\nreturn syscall(SYS_gettid);\n#else\nreturn 0;\n#endif").
                         returns("int").
                         doc("returns the PID for this task")
+
+                    if extended_state_support?
```

Comment:
Mmmm ... Fair enough. What about using orogen_metadata then ? That would require depending on it (i.e. the export would be done by the orogen_metadata plugin itself)

I really want to keep the base interface as clean as possible. Since what you export here *is* metadata, that would make a lot more sense.